code/obj/item/gun/energy.dm

```diff
@@ -1839,3 +1839,428 @@ TYPEINFO(/obj/item/gun/energy/wasp)
 		set_current_projectile(new/datum/projectile/energy_bolt/dazzler)
 		projectiles = list(current_projectile)
 		..()
+
+// Makeshift Laser Rifle
+#define HEAT_REMOVED_PER_PROCESS 15
+#define FIRE_THRESHOLD 125
+TYPEINFO(/obj/item/gun/energy/makeshift)
+	mats = 0
+
+/obj/item/gun/energy/makeshift
+	name = "makeshift laser rifle"
+	icon = 'icons/obj/items/guns/energy64x32.dmi'
+	wear_image_icon = 'icons/mob/clothing/back.dmi'
+	icon_state = "makeshift-energy"
+	item_state = "makeshift_laser"
+	wear_state = "makeshift_laser"
+	w_class = W_CLASS_BULKY
+	c_flags = ONBACK
+	cell_type = null
+	can_swap_cell = FALSE
+	rechargeable = FALSE
+	force = 7
+	two_handed = TRUE
+	can_dual_wield = FALSE
+	desc = "A laser rifle cobbled together from various supplies found around the station, probably not the most reliable weapon in a firefight."
+	muzzle_flash = "muzzle_flash_phaser"
+	charge_icon_state = "laser"
+	spread_angle = 10
+	shoot_delay = 5 DECI SECONDS
+	///What light source we use for the rifle
+	var/obj/item/light/tube/our_light
+	///What battery this gun uses
+	var/obj/item/cell/our_cell
+	///How much heat this weapon has after firing, the weapon breaks if this gets too high
+	var/heat = 0
+	///What step of repair are we on if we have broken? 0 = functional
+	var/heat_repair = 0
+	///What projectile we switch to when update_proj() is called
+	var/lens_proj = /datum/projectile/laser/makeshift
```

Comment:
Why is this necessary given the existing members on this type?

Reply:
fixed